google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryRetryAlgorithm.java

```diff
@@ -92,13 +92,24 @@ public boolean shouldRetry(
   }
 
   private boolean shouldRetryBasedOnBigQueryRetryConfig(
-      Throwable previousThrowable, BigQueryRetryConfig bigQueryRetryConfig) {
+      Throwable previousThrowable, BigQueryRetryConfig bigQueryRetryConfig, ResponseT previousResponse) {
     /*
     We are deciding if a given error should be retried on the basis of error message.
     Cannot rely on Error/Status code as for example error code 400 (which is not retriable) could be thrown due to rateLimitExceed, which is retriable
      */
-    String errorDesc;
-    if (previousThrowable != null && (errorDesc = previousThrowable.getMessage()) != null) {
+    String errorDesc = null;
+    if (previousThrowable != null ){
+      errorDesc = previousThrowable.getMessage();
+    } else if (previousResponse != null){
+      /*
+      In some cases error messages may come without an exception
+      e.g. status code 200 with a rate limit exceeded for job create
+      in these cases there is now previousThrowable so we need to check previousResponse
+       */
+      errorDesc = previousResponse.toString();
```

Comment:
Please add a mock test in `BigQueryImplTest` to verify the correctness of the behavior; i.e. retrying with status code 200 on rateLimitExceeded error message.

Reply:
added mock test to `BigQueryImplTest` test is called `testCreateJobFailureShouldRetry`